[dynamo] actually disable torch.compile'd functions nested in torch._dynamo.disable regions

williamwen42 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #132926
* __->__ #132925

Initial fix for https://github.com/pytorch/pytorch/issues/123771.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

williamwen42 committed:
[dynamo] actually disable torch.compile'd functions nested in torch._dynamo.disable regions

[ghstack-poisoned]

anijain2305 reviewed (commented):
The code LGTM. I will review again when the CI is green.

williamwen42 committed:
Update on "[dynamo] actually disable torch.compile'd functions nested in torch._dynamo.disable regions"


Initial fix for https://github.com/pytorch/pytorch/issues/123771.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

williamwen42 commented:
Tests cannot pass without `enable`. Need to squish changes into https://github.com/pytorch/pytorch/pull/132926.
